design-docs/handling-encodings.md

```diff
@@ -0,0 +1,93 @@
+the content (otherwise, the bytes of the source files are copied as is to the destination). This allows filtering the content 
+correctly from and to the file system, as well as from and to archives. 
+
+For the Zip, War, Ear and Tar tasks, we should add support a `fileNameEncoding` encoding option, that should thus be set on their
+parent AbstractArchiveTask.
+
+To be easier to specify in the build file, these properties should be of type `String`, and not of type `Charset`.
+
+## Testing Plan
+
+We need to add tests which cover each of these scenarios:
+
+- The default platform encoding is honored when either of these properties is not set.
+- When `contentEncoding` is set to something other than the default platform encoding, it is honored.
+- When `fileNameEncoding` is set to something other than the default platform encoding, it is honored.
+- When a character cannot be properly converted into the specified encoding, a clear error message is displayed to the user.
+
+## Documentation Plan
+
+- The CopySpec interface documentation should document the `contentEncoding` property and specify the default value if the property is not set.
+  Since the Copy task inherits from CopySpec, that should document the task automatically.
+- The AbstractArchiveTask should document the `fileNameEncoding` property and specify the default value if the property is not set.
+
+## Open Questions
+
+- The Zip, War and Ear tasks are already using `encoding` as the name of its setting which really deals with `fileNameEncoding`. The best option
+  should be to deprecate this property, make the setter set the new `fileNameEncoding` property, and the getter return the `fileNameEncoding` 
+  property.
+- The name `encoding` is much more common practice in things like the xml standards, ant script, editor configurations, etc. 
+  but it's less clear than `contentEncoding`, and is in conflict with the existing `encoding` property of the Zip, War and Ear tasks. Using
+  it would make it impossible to stay backward-compatible.
+- Should the property accept a null value (as it does in the Zip task) and be null by default, and/or should it be set to the default charset?
+- Are there other existing tasks which should deal with character encoding in a better way?
+- Is it a wise choice to use the platform encoding as the default file name encoding? 
+    - The JDK ZipOutputStream defaults to UTF8
+    - Apache comons ZipArchiveOutputStream defaults to UTF8
+    - Apache Ant's ZipOutputStream, which is used by Gradle now, defaults to the platform default encoding, and changing the default would be a  
+      backward incompatibility.
+- Should the Tar task really allow specifying an encoding, and should it default to the platform encoding? It defaults to ASCII, and [the standard](http://www.gnu.org/software/tar/manual/html_node/Standard.html) doesn't seem to allow anything else. 
```

Comment:
> When creating Zip archives with the Zip task, there should be a metadataCharset property which determines the Charset which will be used to create the file entries and header data in the Zip archive.

Did you see my note about this before? The zip specification basically says utf8 or ascii (technically an older encoding that is not ascii but shares the same visible codepoints). You should just stick with utf8.